apps/src/p5lab/spritelab/commands/customLessonCommands.js

```diff
@@ -0,0 +1,56 @@
+import * as coreLibrary from '../coreLibrary';
+
+export const commands = {
+  // Haiku lesson in CS connections pilot 2021
+  drawHaiku() {
+    const haiku = coreLibrary.screenText.haiku;
+    if (!haiku) {
+      return;
+    }
+    this.textAlign(this.CENTER, this.CENTER);
+    this.stroke('white');
+    this.fill('black');
+    this.strokeWeight(2);
+    this.textFont('Courier New');
+    this.textStyle(this.BOLD);
+    const defaultSize = 25;
+
+    const scaleTextSize = (scaleMultiplier, text) => {
+      const oldTextSize = this.textSize();
+      this.textSize(defaultSize * scaleMultiplier);
+      const size = Math.min(
+        defaultSize * scaleMultiplier,
+        (defaultSize * scaleMultiplier * 370) / this.textWidth(text)
+      );
+      this.textSize(oldTextSize);
```

Comment:
where does `this.textWidth` come from? it would be nice if we could create a modified method that accepts `text` and `size` as params so that we don't have to call `this.textSize` before and after calling `this.textWidth`